Scripts/devEnv.js

```diff
@@ -1,162 +1,240 @@
+ * @param {String} css the root folder of the css files to be included in the html.
+ * @param {String} as the root folder of the assembly scripts files to be compiled.
+ */
+
+const {ESLint} = require("eslint");
+const fs = require("fs");
+const path = require("path");
+
+
+/** This function will find every file in the given directory and his subdirectories and return an array of all the file names.
+ *
+ * @param {String} dir the name of the directory to iterate through.
+ */
+async function getAllFiles(dir) {
+    return fs.promises.readdir(dir).then(async files => {
+        const allFiles = [];
+
+        // For each element we check if it is a directory or a file.
+        // If it is a file then we add it to the array and return nothing, the promise.all will stop at the enclosing .then.
+        // If it is a directory we call the function again with the new directory
+        // We then return the new promse which once resolved will add all files to the array.
+        await Promise.all(
+            files.map(file => {
+                return fs.promises.stat(path.join(dir, file))
+                    .then(stats => stats.isDirectory())
+                    .then(isDir => {
+                        if (isDir) {
+                            return getAllFiles(path.join(dir, file)).then(files => files.concat(allFiles));
+                        } else {
+                            allFiles.push(path.join(dir, file));
+                        }
+                    });
+            })
+        );
+
+        return allFiles;
+    });
+}
+
+(async() => {
+    "use strict;";
+
+    // TODO: change to assemblyscript/asc when this shit get fixed + move it back to the beginning of the file.
+    const asc = await import("assemblyscript/dist/asc.js");
+
+    // 4 arguments + two string that are always there : "node" and the script file name
+    if (process.argv.length < 4 + 2) {
+        console.log("Usage: node devEnv.js <html> <js> <css> <as>");
+        process.exit(1);
+    }
+
+    const START = Date.now();
+    const BUILD_DIR = "build";
+    const LOG_DIR = path.join(BUILD_DIR, "log");
+    const DEV_DIR = path.join(BUILD_DIR, "dev");
+
+    await Promise.all([
+        fs.promises.mkdir(BUILD_DIR, {recursive: true}),
+        fs.promises.mkdir(LOG_DIR, {recursive: true}),
+        fs.promises.mkdir(DEV_DIR, {recursive: true})
+    ]);
+
+    const LOG = await fs.promises.open(
+        path.join(
+            LOG_DIR,
+            new Date(Date.now())
+                .toISOString()
+                .replace(/:/g, "_") +
+                "_dev_env_setup.log"),
+        "w"
+    );
+    const BUFFER = [];
+
+    try {
+        const HTML_SOURCES = (await getAllFiles(process.argv[2])).filter(file => file.endsWith(".html"));
+
+        // We use path.relative to remove the root folder from the file name.
+        const JS_SOURCES = (await getAllFiles(process.argv[3])).map(file => path.relative(process.argv[3], file));

```

Comment:
const JS_SOURCES = (await getAllFiles(process.argv[3])).map(file => path.relative(process.argv[3], file)).filter(file => file.endsWith(".js"));

Reply:
it probably should be the other way ie the others without the filter. I filtered in dev because I added config files to those folder and later removed those  files. there is no more any reason to filter I think

Reply:
I also think that it must be good without it now